File: app/src/main/java/com/example/fondodepantalla/FragmentosAdministrador/ListaAdmin.java
package com.example.fondodepantalla.FragmentosAdministrador;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.fondodepantalla.R;
import com.example.fondodepantalla.Task;
import com.example.fondodepantalla.TaskAdapter;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.ArrayList;
import java.util.List;

public class ListaAdmin extends Fragment {

    private RecyclerView recyclerTasks;
    private TaskAdapter adapter;
    private List<Task> taskList = new ArrayList<>();
    private FirebaseFirestore db;
    private FirebaseAuth auth;

    public ListaAdmin() {}

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_lista_admin, container, false);

        recyclerTasks = view.findViewById(R.id.recyclerTasks);
        recyclerTasks.setLayoutManager(new LinearLayoutManager(getContext()));

        db = FirebaseFirestore.getInstance();
        auth = FirebaseAuth.getInstance();

        adapter = new TaskAdapter(taskList, db, task -> {
            Toast.makeText(requireContext(), "Click: " + task.getId(), Toast.LENGTH_SHORT).show();

            Bundle args = new Bundle();
            args.putString("taskId", task.getId());

            RegistrarAdmin registrarAdmin = new RegistrarAdmin();
            registrarAdmin.setArguments(args);

            ((androidx.appcompat.app.AppCompatActivity) requireActivity())
                    .getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.fragment_containerA, registrarAdmin)
                    .addToBackStack(null)
                    .commit();
        });

        recyclerTasks.setAdapter(adapter);

        String uidActual = auth.getCurrentUser() != null ? auth.getCurrentUser().getUid() : null;

        if (uidActual != null) {
            db.collection("tareas")
                    .whereEqualTo("uidAsignado", uidActual) // <-- Filtra por el UID
                    .addSnapshotListener((value, error) -> {
                        if (error != null) return;
                        taskList.clear();
                        if (value != null) {
                            for (DocumentSnapshot doc : value.getDocuments()) {
                                Task t = doc.toObject(Task.class);
                                if (t != null) {
                                    t.setId(doc.getId());
                                    Boolean completada = doc.getBoolean("confirmacionExito");
                                    if (completada == null || !completada) {
                                        taskList.add(t);
                                    }
                                }
                            }
                            adapter.notifyDataSetChanged();
                        }
                    });
        } else {
            Toast.makeText(requireContext(), "Error: no se encontró UID del usuario", Toast.LENGTH_SHORT).show();
        }

        return view;
    }
}

File: app/src/main/java/com/example/fondodepantalla/FragmentosAdministrador/ListaAdmin.kt
package com.example.fondodepantalla.FragmentosAdministrador

import android.os.Bundle
import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import androidx.fragment.app.Fragment
import com.example.fondodepantalla.R
import com.example.fondodepantalla.Task
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore

class ListaAdmin : Fragment() {

    override fun onCreateView(
        inflater: android.view.LayoutInflater,
        container: android.view.ViewGroup?,
        savedInstanceState: Bundle?
    ): android.view.View {
        return ComposeView(requireContext()).apply {
            setContent { ListaAdminComposable() }
        }
    }

    @Composable
    fun ListaAdminComposable() {
        val firestore = FirebaseFirestore.getInstance()
        val auth = FirebaseAuth.getInstance()
        val context = LocalContext.current
        var tasks by remember { mutableStateOf(listOf<Task>()) }

        LaunchedEffect(Unit) {
            val uidActual = auth.currentUser?.uid
            if (uidActual != null) {
                firestore.collection("tareas")
                    .whereEqualTo("uidAsignado", uidActual)
                    .addSnapshotListener { snapshot, error ->
                        if (error != null) return@addSnapshotListener

                        val lista = snapshot?.documents?.mapNotNull { doc ->
                            val t = doc.toObject(Task::class.java)?.apply { id = doc.id }
                            if (t != null &&
                                !(doc.getBoolean("confirmacionExito") == true &&
                                        !doc.getString("firmaCliente").isNullOrEmpty() &&
                                        (doc.getLong("progress") ?: 0L) == 4L)
                            ) t else null
                        } ?: emptyList()

                        tasks = lista
                    }
            } else {
                Toast.makeText(context, "Error: no se encontró UID del usuario", Toast.LENGTH_SHORT).show()
            }
        }

        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(8.dp)
        ) {
            item {
                Text(
                    text = "Nuevos encargos",
                    style = MaterialTheme.typography.titleMedium,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 16.dp),
                    color = Color.Black
                )
            }

            items(tasks) { task ->
                TaskCard(task = task, firestore = firestore) { tarea ->
                    val fragment = RegistrarAdmin()
                    fragment.arguments = Bundle().apply { putString("taskId", tarea.id) }
                    requireActivity().supportFragmentManager.beginTransaction()
                        .replace(R.id.fragment_containerA, fragment)
                        .addToBackStack(null)
                        .commit()
                }
            }
        }
    }

    @Composable
    fun TaskCard(
        task: Task,
        firestore: FirebaseFirestore,
        onTaskClick: (Task) -> Unit
    ) {
        val context = LocalContext.current
        var isTaken by remember { mutableStateOf(task.isTaken) }
        val uid = FirebaseAuth.getInstance().currentUser?.uid

        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 6.dp, horizontal = 12.dp)
                .clickable { if (task.user == uid) onTaskClick(task) },
            elevation = CardDefaults.cardElevation(defaultElevation = 8.dp)
        ) {
            Row(
                modifier = Modifier
                    .background(Color.White)
                    .padding(16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    painter = painterResource(id = R.drawable.ic_task),
                    contentDescription = null,
                    modifier = Modifier.size(48.dp),
                    tint = Color.Unspecified
                )
                Spacer(modifier = Modifier.width(16.dp))

                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = task.name ?: "Sin nombre",
                        style = MaterialTheme.typography.bodyLarge,
                        color = Color.Black
                    )
                    Text(
                        text = task.descripcion ?: "Sin descripción",
                        style = MaterialTheme.typography.bodyMedium,
                        color = Color.Gray,
                        maxLines = 2
                    )
                }

                Switch(
                    checked = isTaken,
                    onCheckedChange = { checked ->
                        if (checked && uid != null) {
                            firestore.collection("tareas").document(task.id!!)
                                .update("taken", true, "user", uid)
                                .addOnSuccessListener {
                                    isTaken = true
                                    onTaskClick(task)
                                }
                                .addOnFailureListener {
                                    Toast.makeText(context, "Error al tomar tarea", Toast.LENGTH_SHORT).show()
                                }
                        } else if (!checked && task.user == uid) {
                            firestore.collection("tareas").document(task.id!!)
                                .update("taken", false, "user", "")
                                .addOnSuccessListener { isTaken = false }
                                .addOnFailureListener {
                                    Toast.makeText(context, "Error al liberar tarea", Toast.LENGTH_SHORT).show()
                                }
                        } else {
                            Toast.makeText(context, "No puedes liberar esta tarea", Toast.LENGTH_SHORT).show()
                            isTaken = true
                        }
                    }
                )
            }
        }
    }
}
